File: frontend/src/types/Vehicle.ts
export const VehicleType = {
  Gasoline: 0,
  Diesel: 1,
  Electric: 2,
  Hybrid: 3,
  PlugInHybrid: 4,
} as const;

export type VehicleType = typeof VehicleType[keyof typeof VehicleType];

export interface Vehicle {
  id: number;
  make: string;
  model: string;
  year: number;
  purchasePrice: number;
  ownershipStart: string;
  ownershipEnd?: string;
  vehicleType: number;
  vehicleTypeDisplay: string;
  userId: string;
}

export interface CreateVehicleDto {
  make: string;
  model: string;
  year: number;
  purchasePrice: number;
  ownershipStart: string;
  ownershipEnd?: string;
  vehicleType: number;
}

export interface UpdateVehicleDto {
  make: string;
  model: string;
  year: number;
  purchasePrice: number;
  ownershipStart: string;
  ownershipEnd?: string;
  vehicleType: number;
}

File: backend/src/VehicleExpenseAPI/Models/Vehicle.cs
using System.ComponentModel.DataAnnotations;

namespace VehicleExpenseAPI.Models;

public class Vehicle
{
    public int Id { get; set; }
    
    [Required]
    [MaxLength(100)]
    public string Make { get; set; } = string.Empty;
    
    [Required]
    [MaxLength(100)]
    public string Model { get; set; } = string.Empty;
    
    [Range(1900, 2100)]
    public int Year { get; set; }
    
    public decimal PurchasePrice { get; set; }
    
    public DateOnly OwnershipStart { get; set; }
    
    public DateOnly? OwnershipEnd { get; set; }
    
    public VehicleType VehicleType { get; set; } = VehicleType.Gasoline;
    
    // Foreign key to User
    [Required]
    public string UserId { get; set; } = string.Empty;
    
    // Navigation property
    public User? User { get; set; }
}
